x-pack/plugins/lens/public/datasources/form_based/operations/definitions/terms/include_exclude_options.tsx

```diff
@@ -99,19 +99,41 @@ export const IncludeExcludeRow = ({
     selectedOptions: IncludeExcludeOptions[],
     operation: 'include' | 'exclude'
   ) => {
+    const otherOperation = operation === 'include' ? 'exclude' : 'include';
+    const otherSelectedOptions = includeExcludeSelectedOptions[otherOperation] ?? [];
+    const hasIdenticalOptions = selectedOptions.some((option) => {
+      return otherSelectedOptions.some((otherOption) => otherOption.label === option.label);
+    });
+
+    const otherSelectedNonIdenticalOptions = hasIdenticalOptions
+      ? otherSelectedOptions.filter(
+          (otherOption) => !selectedOptions.some((option) => option.label === otherOption.label)
+        )
+      : otherSelectedOptions;
+
     const options = {
-      ...includeExcludeSelectedOptions,
+      [otherOperation]: otherSelectedNonIdenticalOptions,
       [operation]: selectedOptions,
     };
     setIncludeExcludeSelectedOptions(options);
-    const terms = selectedOptions.map((option) => {
-      if (!Number.isNaN(Number(option.label))) {
-        return Number(option.label);
-      }
-      return option.label;
-    });
+
+    const getTerms = (updatedSelectedOptions: IncludeExcludeOptions[]) =>
+      updatedSelectedOptions.map((option) => {
+        if (!Number.isNaN(Number(option.label))) {
+          return Number(option.label);
+        }
+        return option.label;
+      });
+
+    const terms = getTerms(selectedOptions);
     const param = `${operation}IsRegex`;
     updateParams(operation, terms, param, false);
+
+    if (hasIdenticalOptions) {
+      const otherTerms = getTerms(otherSelectedNonIdenticalOptions);
+      const otherParam = `${otherOperation}IsRegex`;
+      updateParams(otherOperation, otherTerms, otherParam, false);
+    }
   };
 
   const onCreateOption = (
```

Comment:
I wonder if instead of duplicating the code between this and the `onChangeIncludeExcludeOptions` you could reuse it? I think this would probably work, WDYT?

```

  const onCreateOption = (searchValue: string, operation: 'include' | 'exclude') => {
    const newOption = { label: searchValue };
    const selectedOptions = [...(includeExcludeSelectedOptions[operation] ?? []), newOption];
    return onChangeIncludeExcludeOptions(selectedOptions, operation);
  };
```

Reply:
Thanks for the review, @mbondyra! I think that's a great suggestion, I’ve gone ahead and implemented it. I also ran the tests locally and it works.